core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2989,6 +2989,9 @@ private[spark] object Utils extends Logging {
       metadata.append(paths(index).toString)
       index += 1
     }
+    if (paths.length > index) {
+      metadata.append(s", ... ${paths.length - index} more")
```

Comment:
This appended info is probably redacted in https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala#L62 and it won't be shown in plan output.

Reply:
`... number more` matches our redact pattern?

Reply:
Oh sorry, I meant truncated by `StringUtils.abbreviate`

Reply:
Ah OK. That looks to be a tricky one now... 

Would we like to apply strict length limit on the metadata location string, so that StringUtils.abbreviate won't touch the  abbreviated form of output? It's a bit odd we try to abbreviate but apply soft limit & add some information to the abbreviated form which leads to exceed the limit, and abbreviate again to break the abbreviated form.

Reply:
I think making the actual metadata data prettier is fine even though it's abbreviate in printed out result.

Reply:
Another idea (simpler) is that just adding the number of paths in prior, and leave it as it is. The tail part could be abbreviated, but we still show the number of paths in total. (It'll be no longer `... N more`. It's like `Total N paths: `.) Is alternative idea OK for everyone?

Reply:
The original PR reduced the size of memory usage, which is fine. I think the current change is okay as well to make sure raw metadata shows that the paths are abbreviate and truncated.

We will just have to make sure it looks pretty in both printed output and raw metadata.

Reply:
I'm okay either way

Reply:
I'm okay either way, too

Reply:
I've reflected the second option. It's probably less beautiful but easily avoid possible glitches. I've also updated the PR title/description.